allensdk/brain_observatory/behavior/data_objects/stimuli/templates.py

```diff
@@ -30,13 +31,21 @@
 
 class Templates(DataObject, StimulusFileReadableInterface,
                 NwbReadableInterface, NwbWritableInterface):
-    def __init__(self, templates: StimulusTemplate):
+    def __init__(self, templates: Dict[str, StimulusTemplate]):
         super().__init__(name='stimulus_templates', value=templates)
+        self.image_template_key = None
+        self.movie_template_key = None
+        for key in templates.keys():
```

Comment:
i don't like that these values could change inside the for loop. Seems like it could create a scenario for a bug if there are e.g. 3 keys? Could you please set these another way?